tests/test_database.py

```diff
@@ -346,7 +351,17 @@ def test_get_connection(self, begin, managed_connector_with_data):
             engine_type = (
                 AsyncEngine if managed_connector_with_data._driver_is_async else Engine
             )
-            assert isinstance(connection, engine_type._trans_ctx)
+
+            if SQLALCHEMY_VERSION.startswith("1."):
+                assert isinstance(connection, engine_type._trans_ctx)
+            else:
+                assert isinstance(
+                    connection,
+                    (
+                        contextlib._GeneratorContextManager,
+                        contextlib._AsyncGeneratorContextManager,
+                    ),
+                )
```

Comment:
this doesn't seem like the best way to test it for sqlalchemy v2 :thinking:

Reply:
If it's a context manager, could you do:

```
with connection as c:
    assert isinstance(c, WhateverTheRealConnectionClassIs)
```

Reply:
Done :+1: